### ISSUE TITLE
Merging our effort on single-file WebP decoder.

### ISSUE THREAD

**Issue by MikuAuahDark** [matanui159/jebp#5]
Hello,

I wrote [simplewebp](https://github.com/MikuAuahDark/simplewebp) 6 months ago for same purpose: To be able to decode WebP without resorting to libwebp and having easiest integration.

So far, the difference between our implementations are:
* Mine only support lossy VP8 (and lossy VP8 in VP8X). Yours only support lossless VP8 (VP8L).
* Mine tires to be compatible with ANSI C, while yours require C99.
* Mine doesn't use SIMD. Yours does.

As for which one API is better, I can't say this for certain. I'll let you think about it.

Unfortunately since mine copies most of Google's libwebp code, it counts as derived program and it's licensed under BSD-3 as opposed to more permissive MIT.

So far, my implementation can decode [this WebP image](https://mikuauahdark.github.io/pwa/awhdhawd/a!.webp) and return bit-exact RGBA data when compared to libwebp.

Perhaps we can exchange code, thoughts, or opinions to support between each other implementations?

**Comment by a740g** [comment]
@MikuAuahDark This is brilliant. I’ve been looking for a `stb_image`-style WebP decoding library for some time now. I am not sure if @matanui159 is working on their implementation. But my humble request would be for both of you to work together to create a combined, feature-complete implementation.

**Comment by PGNetHun** [comment]
Hi @MikuAuahDark ,
That's great, that you have implemented a lossy WebP decoder!
I've tested it with some images, it really works, colors are also correct, so thank you!
About merging code with JebP:
It seems that this repo is NOT maintained any more (for more than 2 years).

But there is an another very simple WebP Lossless decoder that works too, and is very small (~600 LOC, but no error checkings)!
https://github.com/fencl/whale

You shoult try to include that code in your decoder, it might be easier due to that is much smaller (+ add error checking).

**Comment by MikuAuahDark** [comment]
Yep, that's good alternative. I ended up using whale as reference for my WebP lossless implementation.
